keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,805 @@
+
+	// Count is used only when the count mode is "ExactCount". Must be greater than zero.
+	// If AllocationMode is ExactCount and this field is not specified, the default is one.
+	//
+	// +optional
+	// +oneOf=AllocationMode
+	Count int64 `json:"count,omitempty" protobuf:"bytes,5,opt,name=count"`
+
+	// AdminAccess indicates that this is a claim for administrative access
+	// to the device(s). Claims with AdminAccess are expected to be used for
+	// monitoring or other management services for a device.  They ignore
+	// all ordinary claims to the device with respect to access modes and
+	// any resource allocations.
+	//
+	// This is an alpha field and requires enabling the DRAAdminAccess
+	// feature gate. Admin access is disabled if this field is unset or
+	// set to false, otherwise it is enabled.
+	//
+	// +optional
+	// +featureGate=DRAAdminAccess
+	AdminAccess *bool `json:"adminAccess,omitempty" protobuf:"bytes,6,opt,name=adminAccess"`
+}
+```
+
+Admin access to devices is a privileged operation because it grants users
+access to devices that are in use by other users. Drivers might also remove
+other restrictions when preparing the device. 
+
+In Kubernetes 1.31 (before this KEP was introduced), an example validating admission policy [was
+provided](https://github.com/kubernetes/kubernetes/blob/4aeaf1e99e82da8334c0d6dddd848a194cd44b4f/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml#L1-L11)
+which restricts access to this option. It is the responsibility of cluster
+admins to ensure that such a policy is installed if the cluster shouldn't allow
+unrestricted access.
+
+Starting in Kubernetes 1.33 (when this KEP was introduced), a validation has been added to the REST storage layer to only authorize `ResourceClaim` or `ResourceClaimTemplate` with `adminAccess: true` requests if their namespace has the `kubernetes.io/dra-admin-access` label to only allow it for users with additional privileges. More time is needed to
+figure out how that should work, therefore the field is placed behind the `DRAAdminAccess` feature gate.
+
+The `DRAAdminAccess` feature gate controls whether users can set the `adminAccess` field to
+true when requesting devices. That is checked in the apiserver. In addition,
+the scheduler will not allocate claims with admin access when the feature is
+turned off, or if the field was set prior to the feature gate was introduced (for example, set in 1.31 when it
+was available unconditionally, or if the field was set while the feature gate was enabled).
+A similar check in the kube-controller-manager prevents creating a
+ResourceClaim when the ResourceClaimTemplate has the `adminAccess` field while the feature gate is turned off.
+
+### REST Storage Changes
+
+In both pkg/registry/resource/resourceclaim/strategy.go and pkg/registry/resource/resourceclaimtemplate/strategy.go, add the following to `Validate` and `ValidateUpdate`:
+
+```go
```

Comment:
I don't think we should have Go code in the KEP. Instead, describe what that code needs to do.